.github/workflows/e2e_scripts/cleanup_monitoring.py

```diff
@@ -0,0 +1,112 @@
+    Returns:
+        str -- the project name
+    """
+    project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+    if not project_id:
+        raise MissingProjectIdError(
+            'Set the environment variable ' +
+            'GOOGLE_CLOUD_PROJECT to your Google Cloud Project Id.')
+    return project_id
+
+def cleanupDashboards():
+    """ Deletes all dashboards. """
+    client = v1.DashboardsServiceClient()
+    dashboards = client.list_dashboards(project_name)
+    for dashboard in dashboards:
+        try:
+            client.delete_dashboard(dashboard.name)
+        except:
+            print('Could not delete dashboard: ' + dashboard.name)
+
+def cleanupPolicies():
+    """ Delete all alerting policies for both uptime checks and SLOs. """
+    client = monitoring_v3.AlertPolicyServiceClient()
+    policies = client.list_alert_policies(project_name)
+    for policy in policies:
+        try:
+            client.delete_alert_policy(policy.name)
+        except:
+            print('Could not delete alerting policy: ' + policy.name)
+
+def cleanupNotificationChannels():
+    """ Deletes all notification channels. """
+    client = monitoring_v3.NotificationChannelServiceClient()
+    channels = client.list_notification_channels(project_name)
+    for channel in channels:
+        try:
+            client.delete_notification_channel(channel.name)
+        except:
+            print('Could not delete notification channel: ' + channel.name)
+
+def cleanupServices():
+    """ Deletes only custom services that are identified by a trailing 'srv' in the name. """
+    client = monitoring_v3.ServiceMonitoringServiceClient()
+    services = client.list_services(project_name)
+    for service in services:
+        if service.name.endswith("srv"): # Only delete the custom services
+            try:
+                client.delete_service(service.name)
+            except:
+                print('Could not delete service: ' + service.name)
+
+def cleanupSlos():
+    """ Deletes every SLO associated with every service. """
+    client = monitoring_v3.ServiceMonitoringServiceClient()
+    services = client.list_services(project_name)
+    for service in services:
+        slos = client.list_service_level_objectives(service.name)
+        for slo in slos:
+            try:
+                client.delete_service_level_objective(slo.name)
+            except:
+                print('Could not delete SLO: ' + slo.name)
+
+def cleanupUptimeCheck():
+    """ Deletes every uptime check. """
+    client = monitoring_v3.UptimeCheckServiceClient()
+    uptime_checks = client.list_uptime_check_configs(project_name)
+    for check in uptime_checks:
+        try:
+            client.delete_uptime_check_config(check.name)
+        except:
+            print('Could not delete uptime check: ' + check.name)
+
+def doCleanup():
+    """ Ensures that resources are deleted in the proper order so no exceptions are thrown. """
```

Comment:
What happens if the resources don't exist? We want to be able to run cleanup before and after each test, so it shouldn't throw an error if the resources aren't there

Reply:
If the resources don't exist then the list of resources to delete are all empty. There isn't an error, the script does nothing and passes.

Reply:
perfect!